microstream-cache/src/main/java/io/micronaut/microstream/cache/MicrostreamCacheConfiguration.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.microstream.cache;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.naming.Named;
+import io.micronaut.core.util.Toggleable;
+
+/**
+ * Microstream Cache Configuration.
+ * @author Sergio del Amo
+ * @since 1.0.0
+ * @param <K> Key Type
+ * @param <V> Value Type
+ */
+public interface MicrostreamCacheConfiguration<K, V> extends Named, Toggleable {
```

Comment:
Again, I'm struggling to see the point of this interface. It would make sense to me if it was used to declare [immutable configuration](https://docs.micronaut.io/latest/guide/#immutableConfig), but the way it is done seems unnecessary IMHO.

Note that for a bunch of config properties there are now 4 classes/interfaces.

Reply:
I think it is good to have a `MicrostreamCacheConfiguration` and a backing configuration properties. If the user wants to create its own implementation of the interface and don't use configuration he can.